luigi/contrib/s3.py

```diff
@@ -163,350 +161,217 @@ def remove(self, path, recursive=True):
             return True
 
         return False
 
+    def get(self, s3_path, destination_local_path):
+        """
+        Get an object stored in S3 and write it to a local path.
+        """
+        (bucket, key) = self._path_to_bucket_and_key(s3_path)
+        # download the file
+        self.s3.meta.client.download_file(bucket, key, destination_local_path)
+
+    def get_as_string(self, s3_path):
+        """
+        Get the contents of an object stored in S3 as a string.
+        """
+        (bucket, key) = self._path_to_bucket_and_key(s3_path)
+
+        # get the content
+        obj = self.s3.Object(bucket, key)
+        contents = obj.get()['Body'].read().decode('utf-8')
+        return contents
+
+    def exists(self, path):
+        """
+        Does provided path exist on S3?
+        """
+        (bucket, key) = self._path_to_bucket_and_key(path)
+
+        # root always exists
+        if self._is_root(key):
+            return True
+
+        # file
+        s3_key = False
+        try:
+            self.s3.Object(bucket, key).load()
+        except botocore.exceptions.ClientError as e:
+            if e.response['Error']['Code'] in ['NoSuchKey', '404']:
+                s3_key = False
+            else:
+                raise
+        else:
+            s3_key = True
+        if s3_key:
+            return True
+
+        if self.isdir(path):
+            return True
+
+        logger.debug('Path %s does not exist', path)
+        return False
+
     def get_key(self, path):
         """
         Returns just the key from the path.
-
         An s3 path is composed of a bucket and a key.
-
         Suppose we have a path `s3://my_bucket/some/files/my_file`. The key is `some/files/my_file`.
         """
         (bucket, key) = self._path_to_bucket_and_key(path)
 
-        s3_bucket = self.s3.get_bucket(bucket, validate=True)
-
-        return s3_bucket.get_key(key)
+        try:
+            self.s3.meta.client.get_object(Bucket=bucket, Key=key)
+        except botocore.exceptions.ClientError as e:
+            if e.response['Error']['Code'] in ['NoSuchKey', '404']:
+                return None
+            else:
+                raise
+        return key
 
     def put(self, local_path, destination_s3_path, **kwargs):
         """
         Put an object stored locally to an S3 path.
 
         :param kwargs: Keyword arguments are passed to the boto function `set_contents_from_filename`
         """
+        if 'encrypt_key' in kwargs:
+            raise DeprecatedBotoClientException(
+                'encrypt_key deprecated in boto3. Please refer to boto3 documentation for encryption details.')
+
         (bucket, key) = self._path_to_bucket_and_key(destination_s3_path)
 
         # grab and validate the bucket
-        s3_bucket = self.s3.get_bucket(bucket, validate=True)
+        if not self.validate_bucket(bucket):
+            self.s3.create_bucket(Bucket=bucket)
```

Comment:
If that functionality is desired, i'd lobby for an option parameter which is off by default.